src/v3/plugins/github/createGraph.js

```diff
@@ -0,0 +1,167 @@
+// @flow
+
+import {Graph, NodeAddress} from "../../core/graph";
+import type {
+  GithubResponseJSON,
+  RepositoryJSON,
+  ReviewJSON,
+  PullJSON,
+  IssueJSON,
+  CommentJSON,
+  ReviewCommentJSON,
+  NullableAuthorJSON,
+} from "./graphql";
+
+import type {
+  RepoAddress,
+  IssueAddress,
+  PullAddress,
+  ReviewAddress,
+  UserlikeAddress,
+  StructuredAddress,
+  AuthorableAddress,
+  ChildAddress,
+  ParentAddress,
+} from "./nodes";
+import {toRaw} from "./nodes";
+
+import {createEdge} from "./edges";
+
+import {
+  reviewUrlToId,
+  issueCommentUrlToId,
+  pullCommentUrlToId,
+  reviewCommentUrlToId,
+} from "./urlIdParse";
+
+export function createGraph(data: GithubResponseJSON): Graph {
+  const creator = new GraphCreator();
+  creator.addData(data);
+  return creator.graph;
+}
+
+class GraphCreator {
+  graph: Graph;
+
+  constructor() {
+    this.graph = new Graph();
+  }
+
+  addNode(addr: StructuredAddress) {
+    this.graph.addNode(toRaw(addr));
+  }
+
+  addData(data: GithubResponseJSON) {
+    this.addRepository(data.repository);
+  }
+
+  addRepository(repoJSON: RepositoryJSON) {
+    const repo: RepoAddress = {
+      type: "REPO",
+      owner: repoJSON.owner.login,
+      name: repoJSON.name,
+    };
+    this.addNode(repo);
+    repoJSON.issues.nodes.forEach((issue) => this.addIssue(repo, issue));
+    repoJSON.pulls.nodes.forEach((pull) => this.addPull(repo, pull));
+  }
+
+  addIssue(repo: RepoAddress, issueJSON: IssueJSON) {
+    const issue: IssueAddress = {
+      type: "ISSUE",
+      repo,
+      number: String(issueJSON.number),
+    };
+    this.addNode(issue);
+    this.addAuthors(issue, issueJSON.author);
+    this.addHasParent(issue, repo);
+    issueJSON.comments.nodes.forEach((comment) =>
+      this.addComment(issue, comment)
+    );
+  }
+
+  addPull(repo: RepoAddress, pullJSON: PullJSON) {
+    const pull: PullAddress = {
+      type: "PULL",
+      repo,
+      number: String(pullJSON.number),
+    };
+    this.addNode(pull);
+    this.addAuthors(pull, pullJSON.author);
+    this.addHasParent(pull, repo);
+    pullJSON.comments.nodes.forEach((comment) =>
```

Comment:
If you change `comment` to `c`, this fits on a line. :-)

(Same with `reviewJSON.comments` below.)

Reply:
Done.